Orange/preprocess/impute.py

```diff
@@ -296,7 +296,10 @@ def __init__(self, variable, distribution):
             self.sample_prob = np.ones_like(counts) / len(counts)
 
     def transform(self, c):
-        c = np.array(c, copy=True)
+        if not sp.issparse(c):
+            c = np.array(c, copy=True)
+        else:
+            c = c.toarray().ravel()
```

Comment:
What about if we take only `c.data` here and we would need to density the whole column? Consequently, we would need to set only `c.data` in L314.